app/code/Magento/CatalogImportExport/Model/Import/Product.php

```diff
@@ -1328,12 +1337,14 @@ protected function _saveProductCategories(array $categoriesData)
             $categoriesIn = [];
             $delProductId = [];
 
+            $productCategoriesData = [];
             foreach ($categoriesData as $delSku => $categories) {
                 $productId = $this->skuProcessor->getNewSku($delSku)['entity_id'];
                 $delProductId[] = $productId;
 
                 foreach (array_keys($categories) as $categoryId) {
                     $categoriesIn[] = ['product_id' => $productId, 'category_id' => $categoryId, 'position' => 0];
+                    $productCategoriesData[$productId][] = $categoryId;
```

Comment:
maybe to assign as key
`$productCategoriesData[$productId][$categoryId] = $categoryId`

so you could use `isset()` and not `in_array()` so it could perform better?